### PR TITLE
DDL: add two limit for JSON column: cannot used as key and cannot have default value.

### PR DESCRIPTION
**Pull request by hicqu**
- [x] `create table x(a json default '{}')` now will fail because json cannot have not null default value.
- [x] `create table x(a json, key(a))` now will fail because json cannot be used as key or index.

### CODE REVIEW

File: ddl/index.go [hunk 1]
```diff
@@ -49,6 +49,11 @@ func buildIndexColumns(columns []*model.ColumnInfo, idxColNames []*ast.IndexColN
 			return nil, errKeyColumnDoesNotExits.Gen("column does not exist: %s", ic.Column.Name)
 		}
 
+		// JSON cannot index.
```
Comment (coocood) [inline ddl/index.go]: JSON column cannot be used in index.

File: ddl/ddl_api.go [hunk 2]
```diff
@@ -988,6 +994,12 @@ func setDefaultAndComment(ctx context.Context, col *table.Column, options []*ast
 			if err != nil {
 				return ErrColumnBadNull.Gen("invalid default value - %s", err)
 			}
+			if value != nil && (col.Tp == mysql.TypeJSON ||
+				col.Tp == mysql.TypeTinyBlob || col.Tp == mysql.TypeMediumBlob ||
+				col.Tp == mysql.TypeLongBlob || col.Tp == mysql.TypeBlob) {
+				// TEXT/BLOB/JSON can't have not null default values.
+				return errBlobCantHaveDefault.GenByArgs(col.Name.O)
+			}
```
Comment (coocood) [inline ddl/ddl_api.go]: Repeated code can be extracted to function.

**Comment by coocood** [comment]
LGTM

File: executor/executor_test.go [hunk 3]
```diff
@@ -989,6 +989,17 @@ func (s *testSuite) TestJSON(c *C) {
 	result = tk.MustQuery(`select a from test_json tj where a = "string"`)
 	result.Check(testkit.Rows(`"string"`))
 
+	// check some DDL limits for TEXT/BLOB/JSON column.
+	var err error
+	_, err = tk.Exec(`create table test_bad_json(a json default '{}')`)
+	c.Assert(err, NotNil)
```
Comment (tiancaiamao) [inline executor/executor_test.go]: You should check the error code.
How can you make sure the error is not table already exists?

**Comment by tiancaiamao** [comment]
rest LGTM

**Comment by hicqu** [comment]
@tiancaiamao ,PTAL again, thanks!

**Comment by tiancaiamao** [comment]
LGTM

**Pull request merged**